### ISSUE TITLE
`rocks spins 2008 EV5` fails

### ISSUE THREAD

**Issue by Siltala** [maxmahlke/rocks#28]
Hi Max,

Checking for spins of 2008 EV5 currently fails with rocks 1.8.11:
```
$ rocks spins 2008 EV5    
Traceback (most recent call last):
  File "/home/lauri/.local/lib/python3.11/site-packages/rocks/core.py", line 948, in __init__
    super().__init__(**ssocard)  # type: ignore
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lauri/.local/lib/python3.11/site-packages/pydantic/main.py", line 165, in __init__
    __pydantic_self__.__pydantic_validator__.validate_python(data, self_instance=__pydantic_self__)
pydantic_core._pydantic_core.ValidationError: 5 validation errors for Rock
spins.period_flag.0
  Input should be a valid string [type=string_type, input_value=3, input_type=int]
    For further information visit https://errors.pydantic.dev/2.3/v/string_type
spins.period_flag.1
  Input should be a valid string [type=string_type, input_value=3, input_type=int]
    For further information visit https://errors.pydantic.dev/2.3/v/string_type
spins.period_flag.2
  Input should be a valid string [type=string_type, input_value=3, input_type=int]
    For further information visit https://errors.pydantic.dev/2.3/v/string_type
spins.period_flag.3
  Input should be a valid string [type=string_type, input_value=3, input_type=int]
    For further information visit https://errors.pydantic.dev/2.3/v/string_type
spins.period_flag.5
  Input should be a valid string [type=string_type, input_value=2, input_type=int]
    For further information visit https://errors.pydantic.dev/2.3/v/string_type

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/lauri/.local/bin/rocks", line 8, in <module>
    sys.exit(cli_rocks())
             ^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/click/core.py", line 1682, in invoke
    cmd_name, cmd, args = self.resolve_command(ctx, args)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/click/core.py", line 1729, in resolve_command
    cmd = self.get_command(ctx, cmd_name)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lauri/.local/lib/python3.11/site-packages/rocks/cli.py", line 45, in get_command
    return echo()
           ^^^^^^
  File "/home/lauri/.local/lib/python3.11/site-packages/rocks/cli.py", line 354, in echo
    rock = core.Rock(id_, datacloud=catalogues)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lauri/.local/lib/python3.11/site-packages/rocks/core.py", line 972, in __init__
    del offending_part[error["loc"][-1]]
        ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
IndexError: list assignment index out of range
```

Following your previous advice, I had a look at the SsODNet datacloud by getting the link through `rocks spin.links 2008ev5`. At least to my eyes, the data on the SsODNet end looks fine in this case.

**Comment by Siltala** [comment]
Also, `2014 OV3` does not error out but does not find data that exists in SsODNet:
``
$ rocks spins 2014 OV3     
No spins on record for 2014 OV3.
```

The raw output from `rocks spin 2014 OV3` on the other hand does find a rotation period from Thirouin+2018a so, evidently the data exists on the SsODNet end but `rocks spins` doesn't return it. Maybe it is the same bug?

**Comment by maxmahlke** [comment]
Fixed in version `1.8.12`. Thanks for reporting!
